lib/reply.js

```diff
@@ -302,6 +307,71 @@ Reply.prototype.code = function (code) {
 
 Reply.prototype.status = Reply.prototype.code
 
+Reply.prototype.getSerializationFunction = function (schemaOrStatus) {
+  let serialize
+
+  if (typeof schemaOrStatus === 'string' || typeof schemaOrStatus === 'number') {
+    serialize = this.context[kSchemaResponse]?.[schemaOrStatus]
+  } else if (typeof schemaOrStatus === 'object') {
+    serialize = this.context[kReplySerializeWeakMap].get(schemaOrStatus)
+  }
+
+  return serialize
+}
+
+Reply.prototype.compileSerializationSchema = function (schema, httpStatus) {
+  const { request } = this
+  const { method, url } = request
+
+  // Check if serialize function already compiled
+  if (this.context[kReplySerializeWeakMap].has(schema)) {
+    return this.context[kReplySerializeWeakMap].get(schema)
+  }
+
+  const serializerCompiler = this.context.serializerCompiler ||
+   this.server[kSchemaController].serializerCompiler ||
+  (
+    // We compile the schemas if no custom serializerCompiler is provided
+    // nor set
+    this.server[kSchemaController].setupSerializer(this.server[kOptions]) ||
+    this.server[kSchemaController].serializerCompiler
+  )
+
+  const serializeFn = serializerCompiler({
+    schema,
+    method,
+    url,
+    httpStatus
+  })
+
+  this.context[kReplySerializeWeakMap].set(schema, serializeFn)
+
+  return serializeFn
+}
+
+Reply.prototype.serializeInput = function (input, schema, httpStatus) {
+  let serialize
+  httpStatus = typeof schema === 'string' || typeof schema === 'number'
+    ? schema
+    : httpStatus
+
+  if (httpStatus != null) {
+    serialize = this.context[kSchemaResponse]?.[httpStatus]
+
+    // TODO: replace for proper error
+    if (serialize == null) throw new FST_ERR_MISSING_SERIALIZATION_FN(httpStatus)
+  } else {
+    // Check if serialize function already compiled
+    if (this.context[kReplySerializeWeakMap].has(schema)) {
```

Comment:
Using an object as a key will cause bugs: developers could change the object post-fact and it will cause problems. You need some sort of deep-equality-hashing for this.
I use http://npm.im/safe-stable-stringify for these kind of purposes. However generating the hash has a cost too.